### PR TITLE
Validate pod's volumes only once and also fix field path in the error message

### PR DESCRIPTION
**Pull request by php-coder**
This is the same change as https://github.com/kubernetes/kubernetes/pull/31927
but for the `SecurityContextConstraints`.

PTAL @liggitt @pweil-
CC @simo5

### CODE REVIEW

File: pkg/security/securitycontextconstraints/provider.go [hunk 1]
```diff
@@ -269,6 +269,25 @@ func (s *simpleProvider) ValidatePodSecurityContext(pod *api.Pod, fldPath *field
 
 	allErrs = append(allErrs, s.sysctlsStrategy.Validate(pod)...)
 
+	// TODO(timstclair): ValidatePodSecurityContext should be renamed to ValidatePod since its scope
```
Comment (liggitt) [inline pkg/security/securitycontextconstraints/provider.go]: nit, remove timstclair

**Comment by liggitt** [comment]
lgtm with nit

File: pkg/security/securitycontextconstraints/provider.go (hunk 1 above)
Comment (php-coder) [reply pkg/security/securitycontextconstraints/provider.go]: Ok, done.

**Comment by php-coder** [comment]
PTAL @soltysh

**Pull request merged**